### PR TITLE
Letsencrypt setup azuredns

### PR DESCRIPTION
**Pull request by bambash**
This PR adds support for AzureDNS in the letsencrypt-setup chart.

### CODE REVIEW

File: incubator/letsencrypt-setup/README.md [hunk 1]
```diff
@@ -62,6 +62,20 @@ The following table lists the configurable parameters of the chart and their def
...
+
+| Parameter | Description |
+| --------- | ----------- |
+| `clientId` | The AAD Service Principal Application ID |
+| `clientSecret` | Name of kubernetes secret that contains the secret Service Principal password |
+| `clientSecretKey` | Key within a kubernetes secret data block that holds the Service Principal password |
+| `subscriptionId` | Azure subscription where the DNS zone lives |
+| `tenantId` | Azure tenant where the subscription lives |
+| `resourceGroupName` | Resource group where the DNS zone lives |
+| `hostedZoneName` | DNS Zone name (ex: example.com) |
```
Comment (sudermanjr) [inline incubator/letsencrypt-setup/README.md:75]: Is it necessary to filter to a single zone per clusterissuer? I for some reason though that cert-manager could manage multiple with a single clusterissuer

**Review by sudermanjr — APPROVED** [review]
LGTM. I will also defer to @lucasreed since he knows this chart a lot better than me. I left one question inline, but it's a non-blocker

File: incubator/letsencrypt-setup/README.md (hunk 1 above)
Comment (bambash) [reply incubator/letsencrypt-setup/README.md:75]: i'm not completely sure, that is a good question. I mainly followed the docs here: https://cert-manager.io/docs/configuration/acme/dns01/azuredns/

Comment (sudermanjr) [reply incubator/letsencrypt-setup/README.md:75]: Looks like you could do multiple solvers, one for each zone you wanted. either way this is fine the way it is.

**Review by lucasreed — APPROVED** [review]
This looks great, thanks for adding it! 👏

**Pull request merged**